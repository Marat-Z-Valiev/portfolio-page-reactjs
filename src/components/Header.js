import React from "react";
import {Navbar, Nav, NavItem, NavDropdown, MenuItem} from "react-bootstrap";

class Header extends React.Component {
	render() {
		const linksMarkup = this.props.links.map((link, index) => {
			return (
				<li className="nav-item active" key={index}>
					<a className="nav-link" href={link.link}>
						{link.label}
					</a>
				</li>
			);
		});
		return (
			<Navbar inverse collapseOnSelect>
				<Navbar.Header>
					<Navbar.Brand>
						<a href="#brand">Marat Valiev</a>
					</Navbar.Brand>
					<Navbar.Toggle />
				</Navbar.Header>
				<Navbar.Collapse>
					<Nav pullRight>
						{linksMarkup}
						{/* <NavItem></NavItem> */}
						{/* <NavItem eventKey={2} href="#">
							Link
						</NavItem>
						<NavDropdown eventKey={3} title="Dropdown" id="basic-nav-dropdown">
							<MenuItem eventKey={3.1}>Action</MenuItem>
							<MenuItem eventKey={3.2}>Another action</MenuItem>
							<MenuItem eventKey={3.3}>Something else here</MenuItem>
							<MenuItem divider />
							<MenuItem eventKey={3.3}>Separated link</MenuItem>
						</NavDropdown> */}
					</Nav>
					{/* <Nav pullRight>
						<NavItem eventKey={1} href="#">
							Link Right
						</NavItem>
						<NavItem eventKey={2} href="#">
							Link Right
						</NavItem>
					</Nav> */}
				</Navbar.Collapse>
			</Navbar>
		);
	}
}

export default Header;
