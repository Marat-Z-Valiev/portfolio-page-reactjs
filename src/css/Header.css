.container {
  background-color: #4594a4;
  font-size: 1.2em;
}
.container h1 {
  color: #ffffff;
  margin-left: 10px;
}
.container .menu {
  display: flex;
  justify-content: space-between;
}
.container .menu .menu-right {
  display: flex;
  align-items: center;
}
.container .menu-list .menu-list-item {
  display: inline-block;
  margin-right: 20px;
}
.container .menu-list .menu-list-item:hover {
  color: #ffffff;
  cursor: pointer;
}
