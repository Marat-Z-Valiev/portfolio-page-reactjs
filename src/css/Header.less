//Navigation bar
.container {
	background-color: #4594a4;
	font-size: 1.2em;

	h1 {
		color: #ffffff;
		margin-left: 10px;
	}

	.menu {
		display: flex;
		justify-content: space-between;

		.menu-right {
			display: flex;
			align-items: center;
		}
	}

	.menu-list {
		.menu-list-item {
			display: inline-block;
			margin-right: 20px;
			&:hover {
				color: #ffffff;
				cursor: pointer;
			}
		}
	}
}
