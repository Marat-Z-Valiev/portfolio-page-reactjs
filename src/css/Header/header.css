.navbar-container {
  background-color: #4594a4;
  font-size: 1.2em;
}
.navbar-container h1 {
  color: #ffffff;
  margin-left: 10px;
}
.navbar-container .menu {
  display: flex;
  justify-content: space-between;
}
.navbar-container .menu .menu-right {
  display: flex;
  align-items: center;
}
.navbar-container .menu-list .menu-list-item {
  display: inline-block;
  margin-right: 20px;
}
.navbar-container .menu-list .menu-list-item:hover {
  color: #ffffff;
  cursor: pointer;
}
.navbar-container .menu-list .active {
  color: #ffffff;
}
.sticky {
  position: fixed;
  top: 0;
  width: 100%;
  z-index: 50;
}
