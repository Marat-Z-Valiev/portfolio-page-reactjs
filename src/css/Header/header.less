.navbar {
	display: flex;
	justify-content: space-between;
	align-items: center;
	background-color: #4594a4;
	font-size: 1.2em;
	height: 80px;
	z-index: 200;

	h1 {
		color: #ffffff;
		margin-left: 10px;
	}
	.nav-list {
		.nav-list-item {
			display: inline-block;
			margin-right: 20px;
		}

		a {
			text-decoration: none;
			&:hover {
				color: #ffffff;
				cursor: pointer;
			}
		}
		.active {
			color: #ffffff;
		}
	}
}

.sticky {
	position: fixed;
	top: 0;
	width: 100%;
	z-index: 50;
}

@media (max-width: 767px) {
	.nav-list {
		display: none;
	}
}

@media (min-width: 768px) {
	.hamburger-button {
		display: none;
	}
}
