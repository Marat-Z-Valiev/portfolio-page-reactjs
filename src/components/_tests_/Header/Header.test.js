import React from "react";
import {shallow} from "enzyme";

import Header from "../../Header";

const component = shallow(<Header />);

describe("Header", () => {
	it("should render correctly", () => {
		expect(component).toMatchSnapshot();
	});
	it("isSticky state should be false", () => {
		expect(component.state().isSticky).toBe(false);
	});
});
