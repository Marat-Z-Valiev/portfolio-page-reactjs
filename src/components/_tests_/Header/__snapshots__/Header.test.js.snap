// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Marat Valiev
        </h1>,
        <button
          aria-hidden="true"
          aria-label="mobile-menu-button"
          className="hamburger-button"
        >
          <i
            className="fa fa-bars fa-3x"
          />
        </button>,
        <ul
          className="nav-list"
        >
          <HeaderLink
            label="Home"
            link="home"
          />
          <HeaderLink
            label="About"
            link="about"
          />
          <HeaderLink
            label="Projects"
            link="projects"
          />
          <HeaderLink
            label="Contact"
            link="contact"
          />
        </ul>,
      ],
      "className": "navbar ",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Marat Valiev",
        },
        "ref": null,
        "rendered": "Marat Valiev",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "aria-hidden": "true",
          "aria-label": "mobile-menu-button",
          "children": <i
            className="fa fa-bars fa-3x"
          />,
          "className": "hamburger-button",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "fa fa-bars fa-3x",
          },
          "ref": null,
          "rendered": null,
          "type": "i",
        },
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <HeaderLink
              label="Home"
              link="home"
            />,
            <HeaderLink
              label="About"
              link="about"
            />,
            <HeaderLink
              label="Projects"
              link="projects"
            />,
            <HeaderLink
              label="Contact"
              link="contact"
            />,
          ],
          "className": "nav-list",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "class",
            "props": Object {
              "label": "Home",
              "link": "home",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "class",
            "props": Object {
              "label": "About",
              "link": "about",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "class",
            "props": Object {
              "label": "Projects",
              "link": "projects",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "3",
            "nodeType": "class",
            "props": Object {
              "label": "Contact",
              "link": "contact",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "ul",
      },
    ],
    "type": "nav",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Marat Valiev
          </h1>,
          <button
            aria-hidden="true"
            aria-label="mobile-menu-button"
            className="hamburger-button"
          >
            <i
              className="fa fa-bars fa-3x"
            />
          </button>,
          <ul
            className="nav-list"
          >
            <HeaderLink
              label="Home"
              link="home"
            />
            <HeaderLink
              label="About"
              link="about"
            />
            <HeaderLink
              label="Projects"
              link="projects"
            />
            <HeaderLink
              label="Contact"
              link="contact"
            />
          </ul>,
        ],
        "className": "navbar ",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Marat Valiev",
          },
          "ref": null,
          "rendered": "Marat Valiev",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "aria-hidden": "true",
            "aria-label": "mobile-menu-button",
            "children": <i
              className="fa fa-bars fa-3x"
            />,
            "className": "hamburger-button",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "fa fa-bars fa-3x",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <HeaderLink
                label="Home"
                link="home"
              />,
              <HeaderLink
                label="About"
                link="about"
              />,
              <HeaderLink
                label="Projects"
                link="projects"
              />,
              <HeaderLink
                label="Contact"
                link="contact"
              />,
            ],
            "className": "nav-list",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "class",
              "props": Object {
                "label": "Home",
                "link": "home",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "class",
              "props": Object {
                "label": "About",
                "link": "about",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "class",
              "props": Object {
                "label": "Projects",
                "link": "projects",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "3",
              "nodeType": "class",
              "props": Object {
                "label": "Contact",
                "link": "contact",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "ul",
        },
      ],
      "type": "nav",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
